### PR TITLE
Attribute Distribution Diagram

### PR DESCRIPTION
**Pull request by tommartensen**

### CODE REVIEW

File: src/Utils/Utils.js [hunk 1]
```diff
@@ -225,6 +225,31 @@ class Utils {
...
+	static getAttributeValueDistribution(events, attributeName) {
+		const attributeValues = [];
+		events.forEach((event) => {
+			event.Attributes.forEach((attribute) => {
+				if (attribute.Name === attributeName) {
+					attributeValues.push(attribute.Value);
+				}
+			});
+		});
+		let distribution = {};
```
Comment (hpibptlab) [inline src/Utils/Utils.js]: MINOR Make "distribution" "const". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS3353)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/AnalyticsView/AnalyticsView.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import {Card, CardText, CardTitle} from "material-ui";
+import ConnectionComponent from "../Utils/ConnectionComponent";
+import config from "../config/config";
+
+class AnalyticsView extends ConnectionComponent {
+	static renderAttributeDistributionDiagram(key, attributeDistribution, attributeName) {
+		return (
+			<Col key={key} md={6}>
+				<Card>
+					<CardTitle title={"Attribute Distribution for " + attributeName}/>
```
Comment (frederike-ramin) [inline src/AnalyticsView/AnalyticsView.js]: Put that into config please.

File: src/AnalyticsView/AnalyticsView.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+						diagrams.push(<div key={key} />);
+					} else {
+						const attributeDistribution = Utils.getAttributeValueDistribution(
+							this.props.events.value, information.attributeName);
+						diagrams.push(
+							AnalyticsView.renderAttributeDistributionDiagram(
+								key, attributeDistribution, information.attributeName));
+					}
+				} else {
+					diagrams.push(<div>The requested diagram cannot be displayed due to lack of data.</div>);
```
Comment (frederike-ramin) [inline src/AnalyticsView/AnalyticsView.js]: config please.

File: src/AnalyticsView/AnalyticsView.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+				}
+			}
+		);
+		return (
+			<div>
+				<Header title={"Analytics"}/>
```
Comment (frederike-ramin) [inline src/AnalyticsView/AnalyticsView.js]: Cooonfig^^

File: src/AnalyticsView/AnalyticsView.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+				<Container className={css(AppStyles.containerMarginTop)}>
+					<Row>
```
Comment (frederike-ramin) [inline src/AnalyticsView/AnalyticsView.js]: Is the row really necessary?

File: src/DetailView/AttributeDistributionDiagram.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+	componentWillReceiveProps(props) {
+		AttributeDistributionDiagram.buildChartDataset(props);
+	}
+
+	componentDidMount() {
+		AttributeDistributionDiagram.buildChartDataset(this.props);
+	}
+
+	static getDiagramLayout() {
```
Comment (frederike-ramin) [inline src/DetailView/AttributeDistributionDiagram.js]: Show only integer pls

File: src/DetailView/AttributeDistributionDiagram.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+				}
+			}
+		}];
+	}
+
+	static buildChartDataset(props) {
+		const {attributeValues, numberOfOccurrences} = props;
+
+		plotly.newPlot(
+			'attributeDistributionDiagram',
```
Comment (frederike-ramin) [inline src/DetailView/AttributeDistributionDiagram.js]: that is the diagramId, right? just use that variable^^

File: src/DetailView/EventTable.js [hunk 8]
```diff
@@ -14,6 +17,13 @@ class EventTable extends Component {
 							<th key={key} 
 								className={css(AppStyles.tableCell, AppStyles.tableHeaderCell)}>
 								{attribute.Name}
+								<a className="showOnHover"
+									href={Utils.getLink(
+										"analytics?eventTypeId=" + this.props.currentEventTypeId + "&entityId="
+										+ this.props.entityId + "&attributeName=" + attribute.Name
+										+ '&types=["attribute-distribution"]')}>
+									<InsertChartButton style={{height: "16px", width: "16px"}}/>
```
Comment (frederike-ramin) [inline src/DetailView/EventTable.js:25]: Why "Insert" chart button?

**Review by frederike-ramin — CHANGES_REQUESTED** [review]

File: src/DetailView/EventTable.js (hunk 8 above)
Comment (tommartensen) [reply src/DetailView/EventTable.js:25]: I cannot do anything about the naming of [this icon](https://material.io/icons/#ic_insert_chart) in Material UI ;)

File: src/AnalyticsView/AnalyticsView.js (hunk 5 above)
Comment (tommartensen) [reply src/AnalyticsView/AnalyticsView.js]: nope

File: src/DetailView/AttributeDistributionDiagram.js (hunk 6 above)
Comment (tommartensen) [reply src/DetailView/AttributeDistributionDiagram.js]: Should be percentage anyway

**Review by frederike-ramin — APPROVED** [review]

**Pull request merged**